impeller/renderer/backend/vulkan/allocator_vk.h

```diff
@@ -27,11 +27,14 @@ class AllocatorVK final : public Allocator {
 
   fml::RefPtr<vulkan::VulkanProcTable> vk_;
   VmaAllocator allocator_ = {};
+  VmaPool staging_buffer_pools_[3] = {};
```

Comment:
3 seems like a reasonable number since it matches the most common swapchain count. In theory with a higher swapchain count a buffer from a pool could still be in use by the time we come back around, but that wouldn't actually cause any issues beyond higher memory use.